PlayKitSmartSwitch/Classes/SmartSwitchMediaEntryInterceptor.swift

```diff
@@ -97,19 +109,31 @@ extension SmartSwitchMediaEntryInterceptor: PKMediaEntryInterceptor {
         request.set { (response: Response) in
             PKLog.debug("Response:\nStatus Code: \(response.statusCode)\nError: \(response.error?.localizedDescription ?? "")\nData: \(response.data ?? "")")
             
-            if let error = response.error {
+            if let error = response.error as? NSError {
+                
+                if let response = response.data as? [String: AnyObject],
+                   let messages = response["messages"] as? [[String: AnyObject]],
+                   let message = messages.first?["message"] as? String,
+                   let reason = messages.first?["code"] as? String {
+                    
+                    let userInfo = [NSLocalizedDescriptionKey: message, NSLocalizedFailureReasonErrorKey: reason]
+                    let nsError = NSError(domain: "com.PlayKit.SmartSwitch",
+                                          code: error.code,
+                                          userInfo: userInfo)
+                    completion(nil, nsError)
+                    return
+                }
+                
                 completion(nil, error)
                 return
             }
```

Comment:
Is this error returned as an NSError for sure? Or do we need to add a case otherwise.

Reply:
Yes it is NSError
https://github.com/kaltura/netkit-ios/blob/9f5c669be770d52b940cad53a106263472fa1a6a/NetKit/Classes/Core/Executor/implementors/KNKRequestExecutor.swift#L150